common/djangoapps/student/management/commands/bulk_change_enrollment.py

```diff
@@ -0,0 +1,101 @@
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+from optparse import make_option
+
+from course_modes.models import CourseMode
+from student.models import CourseEnrollment
+from xmodule.modulestore.django import modulestore
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class Command(BaseCommand):
+    """Management command to change many user enrollments at once."""
+
+    help = """
+    Change the enrollment status for all users enrolled in a
+    particular mode for a course. Similar to the change_enrollment
+    script, but more useful for bulk moves.
+
+    Example:
+
+    Change enrollment for all audit users to honor in the given course.
+        $ ... bulk_change_enrollment -c course-v1:SomeCourse+SomethingX+2016 --from audit --to honor --commit
+
+    Without the --commit option, the command will have no effect.
+    """
+
+    option_list = BaseCommand.option_list + (
+        make_option(
+            '-f', '--from_mode',
+            dest='from',
+            default=None,
+            help='move from this enrollment mode'
+        ),
+        make_option(
+            '-t', '--to_mode',
+            dest='to',
+            default=None,
+            help='move to this enrollment mode'
+        ),
+        make_option(
+            '-c', '--course',
+            dest='course',
+            default=None,
+            help='the course to change enrollments in'
+        ),
+        make_option(
+            '--commit',
+            action='store_true',
+            dest='commit',
+            default=False,
+            help='display what will be done without any effect'
+        )
+    )
+
+    def handle(self, *args, **options):
+        course_id = options.get('course')
+        from_mode = options.get('from_mode')
+        to_mode = options.get('to_mode')
+        commit = options.get('commit')
+
+        if course_id is None:
+            raise CommandError('No course ID given.')
+        if from_mode is None or to_mode is None:
+            raise CommandError('Both `from` and `to` course modes must be given.')
+
+        try:
+            course_key = CourseKey.from_string(course_id)
+        except InvalidKeyError:
+            try:
+                course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+            except InvalidKeyError:
+                raise CommandError('Course ID {} is invalid.'.format(course_id))
+
+        if modulestore().get_course(course_key) is None:
+            raise CommandError('The given course {} does not exist.'.format(course_id))
+
+        if CourseMode.mode_for_course(course_key, to_mode) is None:
+            raise CommandError('The given mode to move users into ({}) does not exist.'.format(to_mode))
+
+        course_key_str = unicode(course_key)
+
+        try:
+            with transaction.atomic():
+                queryset = CourseEnrollment.objects.filter(course_id=course_key, mode=from_mode)
```

Comment:
2.52 seconds course id
2.59 seconds course id and mode = honor
1.67 seconds course id and mode = audit


For fun, I ran some queries against the read replica.

First off, I ran

```sql
select course_id, mode, count(*) from student_courseenrollment group by course_id, mode order by count(*) desc limit 1;
```

to find the course with the highest enrollment. Unsurprisingly, it's CS50. I ran a few queries with different parameters to test things:

```sql
select * from student_courseenrollment where course_id='HarvardX/CS50x3/2015';
```
took 2.52 seconds. Adding `and mode='honor'` took 2.59. `mode='audit'`, which returns the empty set, took 1.67 seconds. `mode='verified'`, which returns 5670 rows, took 1.68 seconds. This makes me think that the command will run in a reasonable amount of time for just about any course run we have.